docs/DeveloperGuide.md

```diff
@@ -1,34 +1,614 @@
+|Operators| To improve productivity while using the system on a daily basis|
+|Maintenance Engineers| To understand how the system was built in order to perform enhancement or re-engineering work|
 
-## User Stories
+#### 1.2.2. Value Proposition
+**TODO {Describe the value proposition: what problem does it solve?}**
+
+### 1.3. Definitions
+**TODO**
+* *HAMS* - Short for Hospital Administrative Management System.
+* *SAM* - Short for Stupendously AwsoMe, an acronym to describe the records module.
+* *BRANDON* - Short for BRillant Ahead of its time Neat Dainty OrigNal, it is an acronym to describe the custom 
+implementation of the save file structure. 
+* *A&D* - Short of Amazing & Dazzling, it is an acronym to describe the command module.
+
+###### [Back to top](#table-of-content)
+
+## 2. Design & Implementation
+### 2.1 Project overview
+HAMS is built using java. Under java's write once run anywhere, HAMS can be built on any platform including Windows,
+MAC-OS and Linux. When running locally on these system, HAMS uses as custom save configuration called 
+BRillant Ahead of its time Neat Dainty OrigiNal (BRANDON) for loading and saving data. This allows HAMS' internal 
+list to be modified by operators before and during run time. 
+
+### 2.2 Module Overview
+The major code can be broken down into modules. The below table is the breakdown of the module's custom
+name and a summarized purpose.
+
+**TODO**
+
+|Module name|Purpose|
+|---------|-------|
+|Records|Contains and provides access to user information|
+|Converter|Formats user input| 
+|BRillant Ahead of its time Neat Dainty OrigiNal (BRANDON) **storage**|?|
+|Amazing & Dazzling (A&D) Commands|Facade classes that deals with input so that different classes can interact with each other|
+|Parser|Parses the user input for command execution|
+
+#### 2.2.1 SAM record module
+
+The record module consists of 2 classes which represent the patients information and appointment details. 
+As a reflection of real world objects, the Patient's class purpose is to store the particulars of a person while the 
+Appointment's class is to store the date-time data. 
+Thus, the rationale of both classes can be grouped as follows:
+>
+> As a reflection of real world entities, to create, store, and retrieve relevant information about the object.
+>
+Following the above purpose, both classes consist of only getter and setter methods. This would ensure a contiguous 
+flow in logical executions as these methods can be called whenever necessary.
+
+##### 2.2.1.1 Process of Object Creation
+Due to the nature of the above classes containing only getter and setter methods, following how the components interact 
+with each other would provide more accuracy in understanding how these classes are called and the role of its 
+methods. 
+To illustrate, the below example is used:
+* editp \index 1 \name Justin \age 69 \Pasir Panjang
+
+![](images/SD_Patient.png)
```

Comment:
![image](https://user-images.githubusercontent.com/31772358/77992722-36fc2800-7359-11ea-92b9-dd24c3958a1a.png)
Consider ending the activation bar for `:EditPatientCommand` at where the return arrow starts?
Also consider adding/omitting return arrows consistently for the whole diagram? For example, there is no return arrow for the last activation bar of `:Storage` but there is a return arrow for the activation bar of `:Patient`